ui/admin/app/hooks/toolAuth/useToolAuthPolling.ts

```diff
@@ -28,29 +32,32 @@ export function useToolAuthPolling(
 		{ refreshInterval }
 	);
 
-	const getInfo = () => {
-		const agentTools = [
-			...(agent?.tools ?? []),
-			...(agent?.availableThreadTools ?? []),
-			...(agent?.defaultThreadTools ?? []),
-		];
-
-		switch (namespace) {
-			case AssistantNamespace.Agents:
-				return { tools: agentTools, toolInfo: agent?.toolInfo };
-			case AssistantNamespace.Workflows:
-				return { tools: workflow?.tools, toolInfo: workflow?.toolInfo };
-			default:
-				return {};
-		}
-	};
-
-	const { tools, toolInfo } = getInfo();
-
-	// when tool credentials are processing the api will respond with { toolInfo: null }
-	// we need to poll until the toolInfo is not null or there are no tools
-	const shouldPoll = !!tools?.length && !toolInfo;
-	if (shouldPoll !== isPolling) setIsPolling(shouldPoll);
-
-	return { toolInfo, isPolling };
+	useEffect(() => {
+		const getInfo = () => {
+			const agentTools = [
+				...(agent?.tools ?? []),
+				...(agent?.availableThreadTools ?? []),
+				...(agent?.defaultThreadTools ?? []),
+			];
+
+			switch (namespace) {
+				case AssistantNamespace.Agents:
+					return { tools: agentTools, toolInfo: agent?.toolInfo };
+				case AssistantNamespace.Workflows:
+					return { tools: workflow?.tools, toolInfo: workflow?.toolInfo };
+				default:
+					return {};
+			}
+		};
+
+		const { tools, toolInfo: toolInfoFromAgent } = getInfo();
+		if (toolInfoFromAgent) setToolInfo(toolInfoFromAgent);
```

Comment:
might need to add a check here though:

```suggestion
		if (toolInfoFromAgent && toolInfoFromAgent !== toolInfo) setToolInfo(toolInfoFromAgent);
```

Reply:
I think it should be fine to always set it though. It doesn't do deepequal right? anyway agent. toolInfo should be source of truth and always set on the toolInfo variable.